code/modules/mob/living/carbon/human/species/station/skrell/skrell.dm

```diff
@@ -61,8 +61,39 @@
 
 	zombie_type = "Skrell Zombie"
 
+/datum/species/skrell/handle_environment_special(var/mob/living/carbon/human/H)
+	var/old_moisture = H.ST.moisture
+	H.ST.moisture = max(MOISTURE_NONE, H.ST.moisture - rand(8, 13)/10)			// Reduce moisture a little.
+	if(H.fire_stacks && H.ST.moisture > MOISTURE_NORMAL)			// Flaming moist skrell can reduce their flame
+		H.fire_stacks--
+		H.ST.moisture -= rand(150, 350)								// Lost a lot of moisture
+	var/pain_limit = (MOISTURE_NORMAL-H.ST.moisture) * (40 / MOISTURE_NORMAL)	// The max halloss damage to do based on current moisture. At 0, this is 30. At 1800 or higher, this is 0.
+	
+	// Add more pain damage if they need it.
+	var/setPain = max(H.getHalLoss(), pain_limit)
+	H.adjustHalLoss(setPain - H.getHalLoss())
```

Comment:
So, you can use setHalLoss() here in this case. If the brute caused in mob/living's setHalLoss() proc is a concern, you can always do one specific to skrell that does not cause brute damage. Adjust() is always going to be additive or subtractive. So setHalLoss(max(setPain - H.getHalLoss(), H.getHalLoss())